discourse_rock/rockcraft.yaml

```diff
@@ -184,36 +185,6 @@ parts:
     source: bundler
     organize:
       "*": srv/discourse/app/.bundle/
-  install-gems:
-    plugin: nil
-    after:
-      - get-discourse
-      - get-saml-plugin-source
-      - get-prometheus-plugin-source
-      - get-data-explorer
-      - apply-patches
-      - get-bundler-config
-    build-packages:
-      - libpq-dev
-      - ruby2.7
-      - ruby2.7-dev
-      - git
-      - libssl-dev
-    override-build: |
-      cd $CRAFT_STAGE/srv/discourse/app
-      gem install -n "bin" bundler
-      # rexml version is forced to avoid conflicting transient dependency issue
-      sed -i 's/rexml (3.2.5)/rexml (3.2.6)/' Gemfile.lock
-      # The following is a fix for https://github.com/discourse/discourse-prometheus/blob/72fff206ba18ad5ca3112fed2f5f0ce6a17ca6f8/plugin.rb#L13C1-L13C105
-      sed -i "s/gem 'prometheus_exporter', .*/gem 'prometheus_exporter', '0.5.0'/g" Gemfile
-      bin/bundle install
-      bin/bundle install --gemfile="plugins/discourse-prometheus/Gemfile"
-      bin/bundle install --gemfile="plugins/discourse-saml/Gemfile"
-    organize:
-      '/var/lib/gems': var/lib/gems
-      'srv/discourse/app/bin/*': srv/discourse/app/bin/
-      'srv/discourse/app/Rakefile': srv/discourse/app/
-      'srv/discourse/app/bundle': srv/discourse/app/
```

Comment:
@gregory-schiano an interesting PR to look at

I'm curios what gems are used for in this case? Is it an integral part of creating an image for discourse?

A guiding principle for rocks is that they should work standalone, not just when they are used with a charm. So when someone makes use of the image generated by the rockfile, it should do something sensible. So if the gems are required for discourse then they really should be part of the rock.

It would be interesting to see what can be done to reduce the size of the image, e.g., using chisel

Reply:
I see your point and it makes sense but considering that this ROCK is being created for the charm and not to provide a Discourse instance, I don't think there is a problem in transferring the install steps to the charm itself since there is even a [Juju event](https://juju.is/docs/sdk/install-event) for that and the benefits are worthy.

Reply:
I'm a bit torn on this: I feel like the workload OCI image doesn't need to be something that is standalone and at the same time I feel like we shouldn't move too much of the ROCKS logic into the charm (but where's the line ? Is this PR already going too far ?)

Reply:
Apologies I didn't get back to this, it got lost a bit in time off and catching up. Whenever we create rocks the guiding principle is that they should be useful on their own. I know we create lots of rocks mostly for charms - eventually these will be published as sources. So similar to snaps etc, whenever we create these kind of artefacts they should not be charm specific